src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx

```diff
@@ -360,66 +360,92 @@ cdef class CombinatorialPolyhedron(SageObject):
 
             sage: TestSuite(sage.geometry.polyhedron.combinatorial_polyhedron.base.CombinatorialPolyhedron).run()
         """
-        data_modified = None
+        self._equations = ()
+        self._far_face_tuple = ()
 
         if isinstance(data, Polyhedron_base):
-            # input is ``Polyhedron``
-            Vrep = data.Vrepresentation()
-            facets = tuple(inequality for inequality in data.Hrepresentation())
-            self._dimension = data.dimension()
-
-            if not data.is_compact():
-                self._bounded = False
-                far_face = tuple(i for i in range(data.n_Vrepresentation()) if not data.Vrepresentation()[i].is_vertex())
-            else:
-                self._bounded = True
-
-            P = data
-            data = data.incidence_matrix()
-
-            # Delete equations
-            if P.n_equations():
-                data_modified = data.delete_columns([e.index() for e in P.equations()])
-            else:
-                data_modified = data
+            self._init_from_polyhedron(data)
+            return
         elif isinstance(data, LatticePolytopeClass):
```

Comment:
if each branch ends with `return`, perhaps better to use `if` instead of `elif`?

Reply:
Done in 5c21a7947c543159d66a3ac7a0c0c871e3bd3241.